framework/TApplication.php

```diff
@@ -698,6 +699,60 @@ public function getModules()
 		return $this->_modules;
 	}
 
+	/**
+	 * Returns a list of application modules of a specific class.
+	 * Modules that have not been loaded yet are loaded.
+	 * @param string $type class name of the modules to look for.
+	 * @param bool $strict should just the class be found or subclasses of the class
+	 * @param bool $first return the first module found, default false.
+	 * @return array|object list of loaded application module of a specific class
+	 * or the first object of a specific class
+	 * @since 4.2.0
+	 */
+	public function getModulesByType($type, $strict = false, $first = false)
+	{
+		$m = [];
+		foreach ($this->_modules as $id => $module) {
+			if ($module === null) {
+				[$moduleClass, $initProperties, $configElement] = $this->_lazyModules[$id];
+				
+				$match = ($moduleClass === $type);
+				if (!$strict && !$match) {
+					$reflect = new ReflectionClass($moduleClass);
+					if ($reflect = $reflect->getParentClass()) {
+						do {
+							if ($reflect->getName() === $type) {
+								$match = true;
+								break;
+							}
+						} while ($reflect = $reflect->getParentClass());
+					}
+				}
+				
+				if ($match) {
```

Comment:
I'd avoid to make lazy modules load here.
We could return an array of module ids, and the caller would then call `Prado::getApplication->getModule($id)` on the module if it needs to access it

Reply:
I'm matching the output of GetModules.